### DISCUSSION TITLE
How do I get resume training to start at last checkpoint or complete model?

### DISCUSSION THREAD

**Discussion by juan9999** [TheLastBen/fast-stable-diffusion#1193] [Q&A]
I'm trying to resume a dreambooth training to see if I had undertrained a model.

It had last stopped at 650, there was a checkpoint saved at 500 and wanted to continue the training but it seems to start from zero. What am I doing wrong?
<img width="1022" alt="Screenshot 2022-12-29 at 11 35 47 AM" src="https://user-images.githubusercontent.com/31065728/209989660-a347359e-1bb8-49a4-b202-b1c827dce920.png">

**Comment by TheLastBen** [comment]
load the session and choose the saved checkpoint, any training after that would be a continuation to the previous training

**Comment by juan9999** [comment]
Thanks, I'm assuming your recommendations are if I had closed my colab session. 

 Within the same colab session, I had:

Trained the mode
Then ran **Test The Trained Model** to test it

I wasn't happy with the results so I decided to train it some more, so I stopped the execution of  **Test The Trained Model**

I then went to the **Start DreamBooth** step,  checked resume training, increased the **UNet_Training_Steps**  to 1000 (from 650), and set **Text_Encoder_Concept_Training_Steps** to zero and restarted that step.

But it did not reuse the last .ckpt file or any of the checkpoints.
<img width="1034" alt="Screenshot 2022-12-29 at 11 52 52 AM" src="https://user-images.githubusercontent.com/31065728/209995040-f56e524f-0353-4287-87e0-34a5976c7981.png">

**Reply by TheLastBen** [reply]
if it writes "Resuming Training" in green, that means it's using the previous model.

**Comment by juan9999** [comment]
ok so just to confirm resume training doesn’t resume training at the last step it restarts training?

or am i misunderstanding and it is resuming (which preciously stopped at step 650) and going for 1000 more steps?

it would be great to have a txt file produced that lists how the metadata model was trained kinda like the metadata for generated images which includes the prompts.

**Reply by TheLastBen** [reply]
Yes even if it starts from 0, as long as it displays "resuming" that means it continues training.

Implementing a metadata file is on the todo list.

**Reply by SorenTruelsen** [reply]
> ok so just to confirm resume training doesn’t resume training at the last step it restarts training?
> 
> or am i misunderstanding and it is resuming (which preciously stopped at step 650) and going for 1000 more steps?
> 
> it would be great to have a txt file produced that lists how the metadata model was trained kinda like the metadata for generated images which includes the prompts.

It's a bit confusing, lets say I train a model 1000 times. Then I resume, the counter will start at 0 and overwrite the already existing checkpoints (200, 400, 600, 800) with the same name but the new model. So, the one called 200 is actually 1200 etc.

**Reply by TheLastBen** [reply]
Yes, if you want to keep the previous models, just move them manually in your gdrive
